platforms/jvm/testing-jvm-infrastructure/src/testFixtures/groovy/org/gradle/api/internal/tasks/testing/junit/JUnitTestClassProcessorData.groovy

```diff
@@ -257,7 +257,10 @@ public class CustomRunnerWithRunMethodThatBreaksAfterRunningSomeTests extends Ru
     }
 
     Description getDescription() {
-        return Description.createSuiteDescription(type)
+        def desc = Description.createSuiteDescription(type)
+        desc.addChild(Description.createTestDescription(type, "ok1"))
```

Comment:
❔ Is this needed?

Reply:
The current design of `JUnitTestEventAdapter` mandates that all parents are known up-front. I think if necessary I could work around that, but this was literally the only test we have that failed on this, so I don't think it's too big of a deal.